### PR TITLE
Bump Kani version to 0.30.0

### PR DESCRIPTION
**Pull request by karkhaz**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,20 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
...
+## [0.30.0]
+
+## What's Changed
+* Add target selection for cargo kani by @celinval in https://github.com/model-checking/kani/pull/2507
+* Generate Multiple playback harnesses when multiple crashes exist in a single harness. by @YoshikiTakashima in https://github.com/model-checking/kani/pull/2496
+* Escape Zero-size types in playback by @YoshikiTakashima in https://github.com/model-checking/kani/pull/2508
+* Do not crash when `rustfmt` fails. by @YoshikiTakashima in https://github.com/model-checking/kani/pull/2511
+* Update Cbmc version  by @celinval in https://github.com/model-checking/kani/pull/2512
+* Upgrade rust toolchain to 2023-04-30 by @zhassan-aws in https://github.com/model-checking/kani/pull/2456
+* De-duplicate same input injections for the same harness. by @YoshikiTakashima in https://github.com/model-checking/kani/pull/2513
```
Comment (zhassan-aws) [inline CHANGELOG.md:16]: Please move this before the update to CBMC and the rust toolchain. We always keep the dependency updates at the end.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,20 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
...
+
+**Full Changelog**: https://github.com/model-checking/kani/compare/kani-0.29.0...test
```
Comment (zhassan-aws) [inline CHANGELOG.md:18]:
```suggestion
**Full Changelog**: https://github.com/model-checking/kani/compare/kani-0.29.0...kani-0.30.0
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,20 @@ This file contains notable changes (e.g. breaking changes, major changes, etc.)
 
 This file was introduced starting Kani 0.23.0, so it only contains changes from version 0.23.0 onwards.
 
+## [0.30.0]
+
+## What's Changed
```
Comment (celinval) [inline CHANGELOG.md:9]: Can you please add this change as well? https://github.com/model-checking/kani/commit/faa4086e6d7be8fa2a363d6c7d7c0f46607a6429

Comment (karkhaz) [reply CHANGELOG.md:9]: done

**Review by celinval — APPROVED** [review]

**Pull request merged**